<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MqttWidget</class>
 <widget class="QWidget" name="MqttWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>733</width>
    <height>533</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MQTT</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QWidget" name="connectionWidget" native="true">
     <property name="maximumSize">
      <size>
       <width>300</width>
       <height>16777215</height>
      </size>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QGroupBox" name="connection">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>160</height>
         </size>
        </property>
        <property name="title">
         <string>Connection</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="uriTitle">
           <property name="text">
            <string>Server URI</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="skTitle">
           <property name="text">
            <string>Device SK</string>
           </property>
          </widget>
         </item>
         <item row="1" column="4">
          <widget class="QPushButton" name="skBrowse">
           <property name="text">
            <string>Browse...</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="2">
          <widget class="QLabel" name="idTitle">
           <property name="text">
            <string>Device ID</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2" rowspan="2" colspan="3">
          <widget class="QLineEdit" name="id"/>
         </item>
         <item row="3" column="0" rowspan="2">
          <widget class="QLabel" name="statusTitle">
           <property name="text">
            <string>Status</string>
           </property>
          </widget>
         </item>
         <item row="4" column="2" colspan="3">
          <widget class="QLineEdit" name="state">
           <property name="styleSheet">
            <string notr="true">QLineEdit{background:transparent}</string>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="5" column="3">
          <widget class="QPushButton" name="connect">
           <property name="text">
            <string>Init</string>
           </property>
          </widget>
         </item>
         <item row="5" column="4">
          <widget class="QPushButton" name="disconnect">
           <property name="text">
            <string>Uninit</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2" colspan="2">
          <widget class="QLineEdit" name="sk">
           <property name="styleSheet">
            <string notr="true">QLineEdit{background:transparent}</string>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="2" colspan="3">
          <widget class="QLineEdit" name="uri"/>
         </item>
         <item row="5" column="0" colspan="3">
          <spacer name="horizontalSpacer_7">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="subscription">
        <property name="title">
         <string>ReceiveBroadcast</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="QWidget" name="topicOption" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QPushButton" name="subAdd">
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="res.qrc">
                <normaloff>:/image/add.gif</normaloff>:/image/add.gif</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="subDelete">
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="res.qrc">
                <normaloff>:/image/remove.gif</normaloff>:/image/remove.gif</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="subClean">
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="res.qrc">
                <normaloff>:/image/clear.gif</normaloff>:/image/clear.gif</iconset>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_9">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>175</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QTableView" name="topics"/>
         </item>
         <item>
          <widget class="QWidget" name="subButtons" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <spacer name="horizontalSpacer_10">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>115</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="subscribe">
              <property name="text">
               <string>ReceiveBroadcast</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="unsubscribe">
              <property name="text">
               <string>UnReceiveBroadcast</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="publication">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>130</height>
         </size>
        </property>
        <property name="title">
         <string>Broadcast</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="1" colspan="5">
          <widget class="QLineEdit" name="topic"/>
         </item>
         <item row="1" column="1" colspan="3">
          <widget class="QComboBox" name="qos">
           <item>
            <property name="text">
             <string>0 - At Most Once</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1 - At Lesat Once</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2 - Exactly Once</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="1" column="4">
          <widget class="QCheckBox" name="retained">
           <property name="text">
            <string>Retained</string>
           </property>
          </widget>
         </item>
         <item row="1" column="5">
          <widget class="QCheckBox" name="hex">
           <property name="text">
            <string>Hex</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1" colspan="5">
          <widget class="QPlainTextEdit" name="message"/>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="topicTitle">
           <property name="text">
            <string>Topic</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="messageTitle">
           <property name="text">
            <string>Message</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="qosTitle">
           <property name="text">
            <string>QoS</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>45</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="3" column="1" colspan="3">
          <widget class="QPushButton" name="file">
           <property name="text">
            <string>File...</string>
           </property>
          </widget>
         </item>
         <item row="3" column="4" colspan="2">
          <widget class="QPushButton" name="publish">
           <property name="text">
            <string>Broadcast</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="historyGroupBox">
     <property name="title">
      <string>History</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QTableWidget" name="historyTable">
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <column>
         <property name="text">
          <string>Event</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Topic</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Message</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>QoS</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Retained</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Time</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections/>
</ui>
